src/containers/TaskRun/TaskRun.js

```diff
@@ -150,215 +161,178 @@ export /* istanbul ignore next */ class TaskRunContainer extends Component {
     return (
       <LogsRoot
         {...(isLogsMaximized
-          ? { container: this.maximizedLogsContainer.current }
+          ? { container: maximizedLogsContainer.current }
           : null)}
       >
         <Log
           toolbar={getLogsToolbar({
             isMaximized: isLogsMaximized,
             stepStatus,
-            taskRun,
-            toggleMaximized:
-              !!this.maximizedLogsContainer && this.toggleLogsMaximized
+            taskRun: run,
+            toggleMaximized: !!maximizedLogsContainer && toggleLogsMaximized
           })}
-          fetchLogs={() => logsRetriever(stepName, stepStatus, taskRun)}
+          fetchLogs={() => logsRetriever(stepName, stepStatus, run)}
           key={stepName}
           stepStatus={stepStatus}
         />
       </LogsRoot>
     );
   }
 
-  toggleLogsMaximized = () => {
-    this.setState(({ isLogsMaximized }) => ({
-      isLogsMaximized: !isLogsMaximized
-    }));
-  };
-
-  handleTaskSelected = (_, selectedStepId) => {
-    const { history, location, match } = this.props;
+  function handleTaskSelected(_, newSelectedStepId) {
     const queryParams = new URLSearchParams(location.search);
 
-    if (selectedStepId) {
-      queryParams.set(STEP, selectedStepId);
+    if (newSelectedStepId) {
+      queryParams.set(STEP, newSelectedStepId);
       queryParams.delete(TASK_RUN_DETAILS);
     } else {
       queryParams.delete(STEP);
       queryParams.set(TASK_RUN_DETAILS, true);
     }
 
-    const currentStepId = this.props.selectedStepId;
-    if (selectedStepId !== currentStepId) {
+    if (newSelectedStepId !== selectedStepId) {
       queryParams.delete(VIEW);
     }
 
     const browserURL = match.url.concat(`?${queryParams.toString()}`);
     history.push(browserURL);
-  };
+  }
 
-  showRerunNotification = value => {
-    this.setState({ showRerunNotification: value });
-  };
+  function showRerunNotification(value) {
```

Comment:
Is the `showRerunNotification` function a bit redundant now? Could just call `setRerunNotification` directly?

Reply:
Absolutely, good call! Updated to call `setRerunNotification` directly